dev/mcal/inc/Nvic.h

```diff
@@ -0,0 +1,96 @@
+/**
+ * @file Nvic.h
+ * @brief
+ * @version 0.1
+ * @date 2024-04-17
+ *
+ * @copyright Copyright (c) 2024
+ *
+ */
+
+#ifndef DEV_MCAL_INC_NVIC_H_
+#define DEV_MCAL_INC_NVIC_H_
+namespace stm32 {
+namespace dev   {
+namespace mcal  {
+namespace nvic   {
+enum interruptID {
+    // Cortex-M Processor Exceptions
+    NonMaskableInt = -14,    /*!< 2 Non Maskable Interrupt */
+    HardFault = -13,         /*!< 3 Cortex-M0 Hard Fault Interrupt */
+    MemoryManagement = -12,  /*!< 4 Cortex-M0 Memory Management Interrupt */
+    BusFault = -11,          /*!< 5 Cortex-M0 Bus Fault Interrupt */
+    UsageFault = -10,        /*!< 6 Cortex-M0 Usage Fault Interrupt */
+    SVCall = -5,             /*!< 11 Cortex-M0 SV Call Interrupt */
```

Comment:
hi @noura36 ,
Follow The enum coding style like make the first letter uppercase e.g. `enum InterruptID` and for members `k` then uppercase letter. `kNonMaskableInt`.

Apply this comment for all enums used 😄

Reply:
> hi @noura36 ,
> Follow The enum coding style like make the first letter uppercase e.g. `enum InterruptID` and for members `k` then uppercase letter. `kNonMaskableInt`.
> 
> Apply this comment for all enums used 😄 

Welcome back😁

Reply:
> hi @noura36 ,
> Follow The enum coding style like making the first letter as an uppercase e.g. `enum InterruptID` and for members `k` then uppercase letter. `kNonMaskableInt`.
> 
> Apply this comment for all enums used 😄 

Welcome back😁